src/mpid/ch4/generic/am/mpidig_am_msg.h

```diff
@@ -77,13 +77,21 @@ MPL_STATIC_INLINE_PREFIX void MPIDIG_recv_copy(void *in_data, MPIR_Request * rre
         /* otherwise if recv size = 0, it is at least a truncation error */
         MPIR_STATUS_SET_COUNT(rreq->status, 0);
     } else if (p->recv_type == MPIDIG_RECV_DATATYPE) {
+        MPI_Aint max_data_size;
```

Comment:
How about rename to `expected_data_size` or `expected_size`. I was confused by the `max` part.

Reply:
It's not `expected_data_size`.  It is `max_data_size`.  We can receive lesser.  That's OK.